### PR TITLE
feat/add context menu

### PR DESCRIPTION
**Pull request by guibzo**
image

1. Open archive location --> if user has already selected an exe it opens the folder based on this exe location. If not, it opens the installation folder path.
2. Change executable path --> Self-explained.
3. Remove installation archives --> Deletes installation folder; the one located on the download path. Keeps other files (such as the folder where the selected exe with the installed game is) intact. Helpful to get rid of unused archives and free disk space. (includes dialog to confirm if user wants to proceed with action).

solve #220 
solve #130 
solve #118 
solve #71 
solve #416
solve #484

partial solve #421
partial solve #468

### CODE REVIEW

File: src/renderer/src/components/sidebar/sidebar.tsx [hunk 1]
```diff
@@ -120,142 +213,205 @@ export function Sidebar() {
...
+          <section className={styles.section}>
+            <small className={styles.sectionTitle}>{t("my_library")}</small>
+
+            <TextField
+              placeholder={t("filter")}
+              onChange={handleFilter}
+              theme="dark"
+            />
+
+            <div className={styles.menu}>
```
Comment (lilezek) [inline src/renderer/src/components/sidebar/sidebar.tsx:274]: Do not change a `<ul>` into a `<div>`. This element includes `<li>` elements.

File: src/renderer/src/hooks/use-download.ts [hunk 2]
```diff
@@ -104,6 +104,15 @@ export function useDownload() {
         dispatch(removeGameFromDeleting(gameId));
       });
 
+  const removeInstallationFolder = (gameId: number) => {
+    window.electron
+      .deleteGameFolder(gameId)
+      .catch(() => {})
```
Comment (lilezek) [inline src/renderer/src/hooks/use-download.ts:110]: Is this catch needed? Should it at least log the error?

File: src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx [hunk 3]
```diff
@@ -96,20 +94,11 @@ export function HeroPanelActions({
...
-      if (game.executablePath) {
-        window.electron.openGame(game.id, game.executablePath);
-        return;
-      }
-
-      if (game?.executablePath) {
-        window.electron.openGame(game.id, game.executablePath);
-        return;
-      }
+    await updateLibrary();
```
Comment (lilezek) [inline src/renderer/src/pages/game-details/hero/hero-panel-actions.tsx:97]: Why is updating the library needed before opening a game?

**Review by lilezek — CHANGES_REQUESTED** [review]

File: src/renderer/src/components/sidebar/sidebar.tsx [hunk 4]
```diff
@@ -33,6 +39,15 @@ export function Sidebar() {
     initialSidebarWidth ? Number(initialSidebarWidth) : SIDEBAR_INITIAL_WIDTH
   );
 
+  const [isContextMenuOpen, setIsContextMenuOpen] = useState(false);
+  const [axilCoordinates, setAxilCoordinates] = useState({
```
[the remark this answers is not in the document]
Comment (xbozo) [reply src/renderer/src/components/sidebar/sidebar.tsx:43]: yes, wrong spelling*

File: src/renderer/src/components/sidebar/sidebar.tsx [hunk 5]
```diff
@@ -70,6 +81,50 @@ export function Sidebar() {
   const cursorPos = useRef({ x: 0 });
   const sidebarInitialWidth = useRef(0);
 
+  const selectGameExecutable = async () => {
+    const { filePaths } = await window.electron.showOpenDialog({
+      properties: ["openFile"],
+      filters: [
+        {
+          name: "Game executable",
```
[the remark this answers is not in the document]
Comment (xbozo) [reply src/renderer/src/components/sidebar/sidebar.tsx:89]: wdym?

**Comment by xbozo** [comment]
adjusted

**Comment by xbozo** [comment]
> > adjusted
> 
> Everything looks good but the transformation from `<ul>` to `<div>`. Can you leave it to be an `<ul>` as it was originally?
> 
> Also, I'm still wondering why is updating the library needed before opening a game. I'm not saying it is wrong, but I don't get it. Could you elaborate?

debugging reasons.
changed already

**Comment by lilezek** [comment]
LGTM

File: src/main/entity/game.entity.ts
Comment (zamitto) [inline src/main/entity/game.entity.ts:1]: No need to have changes on this file. Keep as little changes as possible, please

File: src/main/events/misc/show-open-dialog.ts
Comment (zamitto) [inline src/main/events/misc/show-open-dialog.ts:1]: No need to have changes on this file. Keep as little changes as possible, please.

And AFAIK, the function used in the event needs to be asyn

File: src/main/events/library/open-game.ts
Comment (zamitto) [inline src/main/events/library/open-game.ts:1]: No need to have changes on this file. Keep as little changes as possible, please

File: src/renderer/src/assets/discord-icon.svg
Comment (zamitto) [inline src/renderer/src/assets/discord-icon.svg:1]: Discord icon is not used anymore

File: src/main/events/library/delete-game-folder.ts
Comment (zamitto) [inline src/main/events/library/delete-game-folder.ts:1]: No need to have changes on this file. Keep as little changes as possible, please.

**Comment by zamitto** [comment]
Are you using Prettier with the project settings? A few changes seems that would be fixed by prettier. And overall, try to make as little changes as possible. I made just a quick review

File: src/renderer/src/assets/discord-icon.svg
Comment (xbozo) [reply src/renderer/src/assets/discord-icon.svg:1]: should sub for telegram?

**Comment by xbozo** [comment]
adjusted

**Pull request closed without merging**